scripts/nbconverted/explore_HighWeight_genes.py

```diff
@@ -0,0 +1,242 @@
+import os
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+from scipy.stats import pearsonr, spearmanr
+
+
+randomState = 123
+from numpy.random import seed
+seed(randomState)
+
+
+# In[3]:
+
+
+# Load arguments
+analysis_name = "sim_AB_2775_300_v2"
+data_file = os.path.join(os.path.dirname(os.getcwd()), "data", analysis_name, "VAE_weight_matrix.txt")
+A_file = os.path.join(os.path.dirname(os.getcwd()), "data", analysis_name, "geneSetA.txt")
+B_file = os.path.join(os.path.dirname(os.getcwd()), "data", analysis_name, "geneSetB.txt")
+
+#HW_file = os.path.join(os.path.dirname(os.getcwd()), "output", analysis_name, "HW_genes.txt")
+
+
+# In[4]:
+
+
+# Read in data
+geneSetA = pd.read_table(A_file, header=0, index_col=0)
+geneSetB = pd.read_table(B_file, header=0, index_col=0)
+
+weight = pd.read_table(data_file, header=0, index_col=0).T
+weight.head(5)
+
+
+# In[5]:
+
+
+# Plot the distribution of gene weights per latent feature
+#num_features = weight.shape[1]
+
+#f, axes = plt.subplots(num_features, 1, sharex=True)
+#for i, ax in zip(weight.columns, axes.flat):
+#    sns.distplot(weight[i] , color="skyblue", rug=False, kde=False, ax=ax)
+    
+#f.set_size_inches(15, 15)
+
+
+# In[6]:
+
+
+# Calculate mean per node ("signature" or "feature")
+means = weight.mean(axis=0)
+
+# Calculate 2 standard deviations per node ("signature" or "feature")
+stds = weight.std(axis=0)
+two_stds = 2*stds
+
+
+# In[7]:
+
+
+# Get high positive and negative weight genes per node ("signature" or "feature") -- HW_df
+# Record metadata per latent feature in a table -- HW_metadata_df
+# Record distance between gene set weight and 
+HW_df = pd.DataFrame()
+HW_metadata_df = pd.DataFrame()
+
+num_nodes = len(means)
+
+for i in range(num_nodes):
+    node_mean = means.iloc[i]
+    node_std = stds.iloc[i] 
+    
+    pos_threshold = node_mean + two_stds.iloc[i]
+    neg_threshold = node_mean - two_stds.iloc[i]
+    
+    hw_pos_genes = weight[weight[i+1] > pos_threshold].index
+    hw_neg_genes = weight[weight[i+1] < neg_threshold].index
+    
+    node = str(i+1)
+    node_name = 'Sig'+node+'pos'
+    
+    # Add high weight positive genes
+    add_pos = pd.DataFrame({node_name: list(hw_pos_genes)})
+    HW_df = pd.concat([HW_df, add_pos], axis=1)
+    
+    # Add metadata for positive node
+    add_metadata_pos = pd.DataFrame({node_name: [node_mean, node_std, pos_threshold]})
+    HW_metadata_df = pd.concat([HW_metadata_df, add_metadata_pos], axis=1)
+    
+    # Add high weight negative genes
+    node_name = 'Sig'+node+'neg'
+    
+    add_neg = pd.DataFrame({node_name: list(hw_neg_genes)})
+    HW_df = pd.concat([HW_df, add_neg], axis=1)
+    
+    # Add metadata for negative node
+    add_metadata_neg = pd.DataFrame({node_name: [node_mean, node_std, neg_threshold]})
+    HW_metadata_df = pd.concat([HW_metadata_df, add_metadata_neg], axis=1)
+
+HW_df = HW_df.T
+HW_metadata_df = HW_metadata_df.T
```

Comment:
Same thing goes with this dataframe. I also think we can avoid the for loop here and just use pandas magic. Something like:

```python
weight = weight.assign(hw_mean=weight.mean(axis=0),
                       hw_std=weight.std(axis=0))

threshold_high_df = weight.hw_mean + (2 * weight.std)
threshold_low_df = weight.hw_mean - (2 * weight.std)

HW_metadata_df = pd.concat([weight.loc[:, ['hw_mean', 'hw_std']],
                            threshold_high_df,
                            threshold_low_df])
```

Reply:
Very helpful tip -- thank you for the advice, I definitely need to learn more of the fancy data structures available in pandas.  Will address in the next PR

Reply:
I think you've made an issue that corresponds with this comment? it is good to link to the issue in the comment (#10 i believe?)

Reply:
Thank you for explaining how to link issues.  This is for issue https://github.com/greenelab/Pseudomonas_latent_spaces/issues/11